File: challenges/tr/solutions/tr.rb
def tr(from, to, str)
  dictionary = {}
  from.size.times { |index| dictionary[from[index]] = to[index] }
  output = ''
  str.each_char { |char| output << (dictionary[char] || char) }
  output
end

p tr("abc","123","dcba")

p tr("abc","123","LOBSTER")

p tr("123", "abc", "bad")

p tr("1a","a1","aax11")

File: challenges/tr/solutions/tr.py
# Submitted by Beverly23


def tr(fromStr, toStr, trstr):
    if not (fromStr and toStr and trstr):
        return trstr
    replace_map = dict(zip(list(fromStr), list(toStr)))
    new_trStr_list = []
    for i in trstr:
        new_trStr_list.append(replace_map.get(i, i))
    return ''.join(new_trStr_list)

# Test cases
print tr("abc","123","dcba")
print tr("abc","123","LOBSTER")
print tr("123", "abc", "bad")
print tr("1a","a1", "aax11")
print tr('abc', 'cab', None)
print tr('abc', 'cab', ' dcba')